scripts/translation.js

```diff
@@ -0,0 +1,90 @@
+const cypherAlphabet = {
+  "A": ".-",
+  "B": "-...",
+  "C": "-.-.",
+  "D": "-..",
+  "E": ".",
+  "F": "..-.",
+  "G": "--.",
+  "H": "....",
+  "I": "..",
+  "J": ".---",
+  "K": "-.-",
+  "L": ".-..",
+  "M": "--",
+  "N": "-.",
+  "O": "---",
+  "P": ".--.",
+  "Q": "--.-",
+  "R": ".-.",
+  "S": "...",
+  "T": "-",
+  "V": "...-",
+  "U": "..-",
+  "W": ".--",
+  "X": "-..-",
+  "Y": "-.--",
+  "Z": "--.."
+};
+
+const cypherNumerals = {
+  "1": ".----",
+  "2": "..---",
+  "3": "...--",
+  "4": "....-",
+  "5": ".....",
+  "6": "-....",
+  "7": "--...",
+  "8": "---..",
+  "9": "----.",
+  "0": "-----"
+};
+
+//english to morse
+export const encoder = (stringToEncode) => {
+
+  if (/[^0-9A-Za-z\s]/.test(stringToEncode)) {
+    throw new Error("Only encodes letters and numbers - don't include special characters")
+  }
+
+  //creates array of uppercase characters and / for spaces
+  const capitalisedStr = stringToEncode.toUpperCase().replaceAll(" ", "/");
+  const messageArr = Array.from(capitalisedStr);
+  
+  //iterates over array to make characters into morse strings
+  let morseArr = [];
+  for (let i = 0; i < messageArr.length; i++) {
+    if (/[A-Z]/.test(messageArr[i])) {
+      morseArr.push(cypherAlphabet[messageArr[i]]);
+    } else if (messageArr[i] === "/") {
+      morseArr.push("/")
+    } else if (/[0-9]/.test(messageArr[i])) {
```

Comment:
this else if logic could be reduced if you have one object with numbers and characters and a regex that checks for chars from a to z and numbers